src/Enhancer/ConditionalEnhancer.php

```diff
@@ -34,11 +39,19 @@ class ConditionalEnhancer implements RouteEnhancerInterface
     private $enhancerMap;
 
     /**
-     * @param array $enhancerMap Ordered list of 'matcher', 'enhancer' pairs
+     * The sorted list of enhancers.
+     *
+     * @var []
+     */
+    private $sortedEnhancerMap = array();
+
+    /**
+     * @param array $enhancerMap Ordered list of 'matcher', 'enhancer' pairs.
+     * @param int   $position    Position to add in the list.
      */
-    public function __construct(array $enhancerMap)
+    public function __construct(array $enhancerMap = array(), $position = 0)
```

Comment:
are you sure we want to make the initial map optional? even if we keep it required, users can still call this with an explicit empty array if they don't know at that point...

on the other hand, why would we need to dynamically add more things to the enhancer after it has been created? i think all of this should be determined at bootstrap. ah, but this map is only for one position...